### PR TITLE
Add `pycoverm`

### PR DESCRIPTION
**Pull request by apcamargo**
This PR adds the [`pycoverm`](https://github.com/apcamargo/pycoverm) Python library.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by apcamargo** [comment]
@prototaxites @mbhall88 I ended up creating a separate PR to include a test. However, I don't think I'll be able to work on this for the next couple of days, so feel free to update the other recipe as well.

`rust-htslib` can be really annoying to build. In my GitHub workflow I have to [install a few dependencies in Ubuntu](https://github.com/apcamargo/pycoverm/blob/98ac587d3482abcac10522f6c7e6b9b30f2365c3/.github/workflows/build_wheels.yaml#L30).

If `pip install` doesn't work, we can try the approach I used in [another recipe](https://github.com/bioconda/bioconda-recipes/blob/master/recipes/pydustmasker/build.sh).

**Comment by apcamargo** [comment]
@BiocondaBot please add label

**Comment by apcamargo** [comment]
@jakobnissen Not sure if this still helps you in any way, but this is finally on a working recipe.

**Review by mbhall88 — APPROVED** [review]

**Comment by mbhall88** [comment]
Actually did you want to add @apcamargo and/or @prototaxites to the recipe maintainers?

**Comment by apcamargo** [comment]
I can be the maintainer. No problem.

Should we put `# [build_platform != target_platform]` in any other build requirement? I'm not familiar with this selector(?)

File: recipes/pycoverm/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+    - {{ compiler("rust") }}
+    - {{ stdlib("c") }}
+    - python                              # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
+    - crossenv                            # [build_platform != target_platform]
+    - maturin >=1.7,<2                    # [build_platform != target_platform]
```
Comment (daler) [inline recipes/pycoverm/meta.yaml:26]: `# [build_platform != target_platform]` is for [cross-compiling](https://conda-forge.org/docs/maintainer/knowledge_base/#cross-compilation) which we are not doing in bioconda, so those can be omitted.

Comment (apcamargo) [reply recipes/pycoverm/meta.yaml:26]: So I guess `crossenv` and `cross-python_{{ target_platform }}` can also be removed?

Comment (daler) [reply recipes/pycoverm/meta.yaml]: I believe so!

**Comment by daler** [comment]
FYI in general our policy is just [simple existence checks](https://conda-forge.org/docs/maintainer/adding_pkgs/#simple-existence-tests) ([bioconda docs](https://bioconda.github.io/contributor/guidelines.html#tests)), but the bam seems small enough so this should be fine.

File: recipes/pycoverm/meta.yaml (hunk 1 above)
Comment (mbhall88) [reply recipes/pycoverm/meta.yaml]: Oh nice. Yeah sorry I just essentially copied that stuff blindly from a conda-forge recipe.

**Pull request merged**